File: goormlevel/175018/1. 수열/수열.cpp
#include <iostream>

#define MOD 1000000007

using namespace std;

int main() {
	int k = 0;
	if (!(cin >> k)) return 0;

	if (k == 1) {
		cout << "0\n";
		return 0;
	}
	else if (k == 2) {
		cout << "1\n";
		return 0;
	}

	long long prev2 = 0;
	long long prev1 = 1;
	long long current = 0;
	for (int i = 3; i <= k; i++) {
		current = (prev1 + prev2) % MOD;
		prev2 = prev1;
		prev1 = current;
	}

	cout << current << "\n";

	return 0;
}

File: goormlevel/175018/1. 수열/수열.py
import sys

MOD = 1000000007

input = sys.stdin.readline
write = sys.stdout.write

def solve():
	k = int(input().rstrip())
	if not k: return

	if k == 1:
		write("0\n")
		return
	elif k == 2:
		write("1\n")
		return

	prev2 = 0
	prev1 = 1
	current = 0
	for i in range(3, k + 1):
		current = (prev2 + prev1) % MOD
		prev2 = prev1
		prev1 = current

	write(str(current) + "\n")

if __name__ == "__main__":
	solve()